bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/SonyAuth.java

```diff
@@ -0,0 +1,393 @@
+        Objects.requireNonNull(transport, "transport cannot be null");
+
+        final String registrationUrl = getRegistrationUrl();
+        if (registrationUrl == null || StringUtils.isEmpty(registrationUrl)) {
+            return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, "No registration URL");
+        }
+
+        // Do the registration first with what the mode says,
+        // then try it again with the other mode (so registration mode sometimes lie)
+        final String[] registrationTypes = new String[2];
+        if (getRegistrationMode() == 2) {
+            registrationTypes[0] = "new";
+            registrationTypes[1] = "initial";
+        } else {
+            registrationTypes[0] = "initial";
+            registrationTypes[1] = "new";
+        }
+
+        final TransportOption[] headers = accessCode == null ? new TransportOption[0]
+                : new TransportOption[] { new TransportOptionHeader(NetUtil.createAuthHeader(accessCode)) };
+        try {
+            final String rqst = "?name=" + URLEncoder.encode(NetUtil.getDeviceName(), "UTF-8") + "&registrationType="
+                    + registrationTypes[0] + "&deviceId=" + URLEncoder.encode(NetUtil.getDeviceId(), "UTF-8");
+            final HttpResponse resp = transport.executeGet(registrationUrl + rqst, headers);
+            if (resp.getHttpCode() != HttpStatus.BAD_REQUEST_400) {
+                return resp;
+            }
+        } catch (final UnsupportedEncodingException e) {
+            // let it got to the next one...
+        }
+
+        try {
+            final String rqst2 = "?name=" + URLEncoder.encode(NetUtil.getDeviceName(), "UTF-8") + "&registrationType="
+                    + registrationTypes[1] + "&deviceId=" + URLEncoder.encode(NetUtil.getDeviceId(), "UTF-8");
+            return transport.executeGet(registrationUrl + rqst2, headers);
+        } catch (final UnsupportedEncodingException e) {
+            return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, e.toString());
+        }
+    }
+
+    /**
+     * Helper method to initiate an IRCC renewal
+     * 
+     * @param transport a non-null transport to use
+     * @return the non-null HttpResponse of the renewal
+     */
+    private HttpResponse irccRenewal(final SonyHttpTransport transport) {
+        Objects.requireNonNull(transport, "transport cannot be null");
+
+        final String registrationUrl = getRegistrationUrl();
+        if (registrationUrl == null || StringUtils.isEmpty(registrationUrl)) {
+            return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, "No registration URL");
+        }
+
+        try {
+            final String parms = "?name=" + URLEncoder.encode(NetUtil.getDeviceName(), "UTF-8")
+                    + "&registrationType=renewal&deviceId=" + URLEncoder.encode(NetUtil.getDeviceId(), "UTF-8");
+            return transport.executeGet(registrationUrl + parms);
```

Comment:
lol - you're now seeing the point I made below.  IRCC authentication is a very strange beast.  I try to authentication with normal scalar methods first (which can be header [psk or requested] or cookie or open).  If that doesn't work, I attempt the IRCC registration (which depends in part on the IRCC registration version and even then you need to try both because different devices use different registration types [the registration version gives us a *hint* on which]).

Basically - IRCC registration URLs are only on old devices (when scalar didn't exist) and generally only on blurays.  Newer devices will implement both - that's why I try scalar first (and is the preferred way).